### PR TITLE
CwlPipeline PrepForTransfer

### PR DESCRIPTION
**Pull request by jasonwalker80**
initial revision of CwlPipeline tool for prepping data transfer.

### CODE REVIEW

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

File: lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    is => 'Command::V2',
+    has => [
+        builds => {
+            is => 'Genome::Model::Build::CwlPipeline',
+            is_many => 1,
+            doc => 'The build(s) to prepare for data transfer',
+        },
+        directory => {
+            is => 'Text',
+            doc => 'The diretory to prepare and write symlinks.',
```
Comment (tmooney) [inline lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm]: `directory` is missing a `c`.

File: lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+use Genome;
+
+class Genome::Model::CwlPipeline::Command::PrepForTransfer {
+    is => 'Command::V2',
+    has => [
+        builds => {
+            is => 'Genome::Model::Build::CwlPipeline',
+            is_many => 1,
+            doc => 'The build(s) to prepare for data transfer',
```
Comment (tmooney) [inline lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm]: Two of the parameter doc lines end in `.` and this one does not.  (I'd probably take them out of the other two, but consistency is good either way :smile:)

File: lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    if ($self->md5sum) {
+        push @headers, 'md5sum';
+    }
+    if (-d $self->directory) {
+        $self->fatal_message('Unable to use existing directory: '. $self->directory);
+    } else {
+        Genome::Sys->create_directory($self->directory);
+    }
+    my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
+        output => $self->directory .'/MANIFEST',
```
Comment (tmooney) [inline lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm]: I'd advocate for using `File::Spec->join` to make paths.

File: lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        separator => "\t",
+        headers => \@headers,
+    );
+    for my $build ($self->builds) {
+        my $results_directory = $build->data_directory .'/results';
```
Comment (tmooney) [inline lib/perl/Genome/Model/CwlPipeline/Command/PrepForTransfer.pm]: This makes me want `CwlPipeline` builds to have a `results_directory` accessor!

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Comment by apipe-tester** [comment]
Merged build finished. Test PASSed.

**Pull request merged**